dialog/file.go

```diff
@@ -251,21 +297,33 @@ func showFileDialog(save bool, callback interface{}, parent fyne.Window) {
 	d.win.Show()
 }
 
+// NewExtensionFileFilter takes a string slice of extensions with a leading . and creates a filter for the file dialog.
+// Example: .jpg, .mp3, .txt, .sh
+func NewExtensionFileFilter(extensions []string) FileFilter {
+	return &extensionFileFilter{extensions: extensions}
+}
+
+// NewMimeTypeFileFilter takes a string slice of mimetypes, including globs, and creates a filter for the file dialog.
+// Example: image/*, audio/mp3, text/plain, application/*
+func NewMimeTypeFileFilter(mimeTypes []string) FileFilter {
+	return &mimeTypeFileFilter{mimeTypes: mimeTypes}
+}
+
 // ShowFileOpen shows a file dialog allowing the user to choose a file to open.
 // The dialog will appear over the window specified.
-func ShowFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window) {
+func ShowFileOpen(callback func(fyne.FileReadCloser, error), parent fyne.Window, filter FileFilter) {
 	if fileOpenOSOverride(callback, parent) {
 		return
 	}
-	showFileDialog(false, callback, parent)
+	showFileDialog(false, callback, parent, filter)
 }
 
 // ShowFileSave shows a file dialog allowing the user to choose a file to save to (new or overwrite).
 // If the user chooses an existing file they will be asked if they are sure.
 // The dialog will appear over the window specified.
-func ShowFileSave(callback func(fyne.FileWriteCloser, error), parent fyne.Window) {
+func ShowFileSave(callback func(fyne.FileWriteCloser, error), parent fyne.Window, filter FileFilter) {
```

Comment:
Given that all usages of this are passing nil in our demos it seems more like an optional parameter.
Perhaps we should update to mirror the design pattern of ShowxxxDialog() being a shortcut for NewxxxDialog().Show() - then the filter could be set on a simple struct representing the dialog creation state?

Reply:
I added a constructor for file and save dialogs that take a FileFilter.  What do you think?